### PR TITLE
S1

### PR DESCRIPTION
**Pull request by Parsley-Afroman**

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+function getDb(): object {
+    return new PDO ('mysql:host=db; dbname=sami-collection','root', 'password');
+}
+
+/**This function takes a database object and selects the items from the database to output an array.
+ * @param object $db is the object from the getDb function that is a connection to the database content.
+ * @return array this returns an array based on the selection criteria within the prepare statement.
+ */
+function getItems(object $db): array {
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
```
Comment (charliecog) [inline functions.php:14]: this can be moved up to the getDb function

File: functions.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
+<?php
+/**A function that connects to the SQL database and retrieves an object.
+ * @return object
+ */
+function getDb(): object {
```
Comment (charliecog) [inline functions.php:5]: can be return type hinted to `PDO` instead object

File: functions.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    return new PDO ('mysql:host=db; dbname=sami-collection','root', 'password');
+}
+
+/**This function takes a database object and selects the items from the database to output an array.
+ * @param object $db is the object from the getDb function that is a connection to the database content.
```
Comment (charliecog) [inline functions.php:10]: please add a line break between the description param and return lines

Comment (charliecog) [reply functions.php:10]: (please do elsewhere in the file too)

File: functions.php [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    $query =  $db->prepare("SELECT `name`, `designer`, `style`, `year_released`, `image` FROM `items` ORDER BY `id`;");
+    $query->execute();
+    return $query->fetchAll();
+}
+
+/**This function receives an array from the database selection and outputs a string that will receive styling.
+ * @param array $items is an array list of the selected criteria from the database query.
+ * @return string for each array entry a string is output that will be displayed in HTML.
+ */
+function arraySplit(array $items) : string{
```
Comment (charliecog) [inline functions.php:24]: a more semantic name would be generateJacketMarkup

Comment (charliecog) [reply functions.php:24]: also $items should be $jacket

File: functions.php [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            if (!isset($item['name'])){
+                $result .= '<div class = "empty">' . 'no name' . '</div>';
+            } else {
+                if (!$item['name']) {
+                    $result .= '<div class = "empty">' . 'there isn\'t a name for this item' . '</div>';
+                } else {
+                    $result .= '<div class = "name">' . $item['name'] . '</div>';
+                }
+            }
+            if (!isset($item['designer'])){
```
Comment (charliecog) [inline functions.php:49]: this isn't needed as your db enforces that this is required

File: functions.php [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+                $result .= '<div class = "empty">' . 'no designer' . '</div>';
+            } else {
+                if (!$item['designer']) {
+                    $result .= '<div class = "empty">' . 'the designer of this garment is not yet known' . '</div>';
+                } else {
+                    $result .= '<div class="designer">' . $item['designer'] . '</div>';
+                }
+            }
+            if (!isset($item['style'])){
```
Comment (charliecog) [inline functions.php:58]: this isn't needed as your db enforces that this is required

File: functions.php [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+                $result .= '<div class = "empty">' . 'no style' . '</div>';
+            } else {
+                if (!$item['style']) {
+                    $result .= '<div class = "empty">' . 'the style of this garment is not yet known' . '</div>';
+                } else {
+                    $result .= '<div class = "style">' . $item['style'] . '</div>';
+                }
+            }
+            if (!isset($item['year_released'])){
```
Comment (charliecog) [inline functions.php:67]: this isn't needed as your db enforces that this is required

**Review by charliecog — COMMENTED** [review]
- year_released in the db field should be a `YEAR` rather than an INT
- unit tests are failing

File: functions.php (hunk 5 above)
Comment (Parsley-Afroman) [reply functions.php]: Have done this

File: functions.php (hunk 7 above)
Comment (Parsley-Afroman) [reply functions.php]: Have done this

File: functions.php (hunk 4 above)
Comment (Parsley-Afroman) [reply functions.php]: Have done this

File: functions.php (hunk 3 above)
Comment (Parsley-Afroman) [reply functions.php]: Have done this

File: functions.php [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    return $db;
+}
+
+/**This takes an object and outputs an array based on the selection criteria passed to the object
+ *
+ * @param object $db is a variable for the object derived from getDb
+ * @return array of items from the array and their associated values
+ */
+function dbSelection(object $db): array {
```
[the remark this answers is not in the document]
Comment (charliecog) [reply functions.php:17]: and type hint can be `PDO`

**Review by charliecog — COMMENTED** [review]
almost there - last couple of changes to get across the line

**Review by charliecog — APPROVED** [review]
nice work!

**Pull request merged**